backend/FwLite/LcmCrdt/RemoteSync/CrdtHttpSyncService.cs

```diff
@@ -42,26 +44,26 @@ public async ValueTask<bool> ShouldSync(ISyncHttp syncHttp)
         return _isHealthy.Value;
     }
 
-    public async ValueTask<ISyncable> CreateProjectSyncable(ProjectData project)
+    /// <summary>
+    /// Creates a Harmony sync client to represent a remote server
+    /// </summary>
+    /// <param name="project">project data, used to provide the projectId and clientId</param>
+    /// <param name="client">should have the base url set to the remote server</param>
+    /// <returns></returns>
+    public async ValueTask<ISyncable> CreateProjectSyncable(ProjectData project, HttpClient client)
     {
-        if (string.IsNullOrEmpty(project.OriginDomain))
-        {
-            logger.LogWarning("Project {ProjectName} has no origin domain, unable to create http sync client", project.Name);
-            return NullSyncable.Instance;
-        }
-
-        var client = await authHelpersFactory.GetHelper(project).CreateClient();
-        if (client is null)
-        {
-            logger.LogWarning("Unable to create http client to sync project {ProjectName}, user is not authenticated to {OriginDomain}", project.Name, project.OriginDomain);
-            return NullSyncable.Instance;
-        }
+        return new CrdtProjectSync(RestService.For<ISyncHttp>(client, refitSettings), project.Id, project.ClientId, this);
+    }
 
-        return new CrdtProjectSync(RestService.For<ISyncHttp>(client, refitSettings), project.Id, project.ClientId  , project.OriginDomain, this);
+    public async ValueTask<bool> TestAuth(HttpClient client)
+    {
+        logger.LogInformation("Testing auth, client base url: {ClientBaseUrl}", client.BaseAddress);
+        var syncable = await CreateProjectSyncable(new ProjectData("test", Guid.Empty, null, Guid.Empty), client);
+        return await syncable.ShouldSync();
```

Comment:
I don't understand the name `ShouldSync`. Should it be `CanSync`?
Under the hood it's doing a health-check, which is implicitly an "I'm authenticated" check.
So, this method is appropriately named `TestAuth`, but the fact that it's actually testing auth is a few layers deep.

So, I feel like `ShouldSync` should be renamed to `CanSync` and `ISyncHttp.HealthCheck` should be renamed to something like `AuthHealthCheck`.

Reply:
That's a fair point, though limited in scope. It's called `ShouldSync` because we could also implement this on the CRDT Client so it returns false if there's no new data to send (right now it always returns true), similarly (though it would require a adding an argument) the server could return false (by calling a different API) when there's no new data to sync. Obviously for now it's quite simple so maybe it should just be `CanSync`

Reply:
`ShouldSync` is a good name for the implementation you're describing 🙂.
Sounds like you have an open todo here that might cover this:
```csharp
//todo pull this out into a service wrapped around auth helpers so that any service making requests can use it
public async ValueTask<bool> ShouldSync(ISyncHttp syncHttp)
```

Reply:
actually that's another issue as that caches the result globally and not per server